subsys/usb/device/class/audio/Kconfig

```diff
@@ -19,4 +19,31 @@ module = USB_AUDIO
 module-str = USB Audio
 source "subsys/logging/Kconfig.template.log_config"
 
+config USB_DEVICE_AUDIO_MIN_VOLUME
+	hex "USB Audio Device Min Volume"
+	default 0xFA00
+	range 0 0xFFFF
+	help
+	  attention: this attribute is a signed value.
+	  This attribute represents the minimum volume level.
+	  The range from +127.9961 dB (0x7FFF) down to -127.9961 dB (0x8001).
+
+config USB_DEVICE_AUDIO_MAX_VOLUME
+	hex "USB Audio Device Max Volume"
+	default 0x0500
+	range 0 0xFFFF
+	help
+	  attention: this attribute is a signed value.
+	  This attribute represents the maximum volume level.
+	  The range from +127.9961 dB (0x7FFF) down to -127.9961 dB (0x8001).
+
+config USB_DEVICE_AUDIO_RES_VOLUME
+	hex "USB Audio Device Resolution Volume"
+	default 0x100
+	range 0x1 0x7FFF
+	help
+	  attention: this attribute can only take positive values.
+	  The range from 1/256 dB (0x0001) to +127.9961 dB (0x7FFF).
+	  0x100 represents 1dB.
+
```

Comment:
Audio class are instantiated using devicetree, but the new Kconfig options affect all instances, even if feature-volume does not exist for a particular instance. Is this the right approach?

Reply:
Thanks. How about I move these configs to dts proprites ?

Reply:
I am not sure how to add some properties dependency `hp-feature-volume`.
How about these change in `dts/bindings/usb/usb-audio-hs.yaml`
```
  hp-feature-volume:
    type: boolean
    description: |
      Enable Volume feature.
  volume_max:
    type: int
    default: 1280
    description: |
      todo: add description
  volume_min:
    type: int
    default: -1536
    description: |
      todo: add description
  volume_res:
    type: int
    default: 256
    description: |
      todo: add description
```

Reply:
Perhaps better to add new `dts/bindings/usb/usb-audio-feature-volume.yaml` and include it in usb-audio-.*. yaml

Reply:
> Perhaps better to add new `dts/bindings/usb/usb-audio-feature-volume.yaml` and include it in usb-audio-.*. yaml

Thanks, I added these into  `dts/bindings/usb/usb-audio-feature-volume.yaml` and include it in  usb-audio-hs. yaml

But I don't know how to get the corresponding "volume_max" by specific instance in audio.c. And I am not sure which struct these properties should be placed in.

Could you give me some tips please?  Thanks.

Reply:
In the drivers we typically have two structures, constant device configuration (dev->config) and runtime device data (dev->data), (unfortunately) audio implementation does not really follow Zephyr Device Driver Model. min, max, and res properties are constant values an likely part of constant device structure, but there is none. You can place min, max, and res in struct usb_audio_dev_data {}, https://github.com/zephyrproject-rtos/zephyr/blob/a18761e29861b709b984eca82dc31efe6214fd62/subsys/usb/device/class/audio/audio.c#L28-L46

or add a pointer to a new constant structure `static const struct feature_volume {uint16_t min; ....};`

Ping me if you are not sure how to get the values from DT properties.

Reply:
Thanks, it looks good, I will try to find time for a detailed review in the next few days.